analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -172,20 +172,68 @@ public void ArrowExpressionBody_WithNotExpectedNode_ReturnsNull()
             ExtensionsCS.GetDeclarationTypeName(SyntaxFactory.StructDeclaration("MyStruct")).Should().Be("struct");
 
         [TestMethod]
-        public void CreateCfg_MethodBody_ReturnsCfg_CS()
+        public void CreateCfg_MethodDeclaration_ReturnsCfg_CS()
         {
-            const string code = @"
-public class Sample
-{
-    public void Main()
-    {
-        var x = 42;
-    }
-}";
+            const string code = """
+                public class Sample
+                {
+                    public void Main()
+                    {
+                        var x = 42;
+                    }
+                }
+                """;
             var (tree, semanticModel) = TestHelper.CompileCS(code);
             var node = tree.Single<SyntaxCS.MethodDeclarationSyntax>();
 
-            ExtensionsCS.CreateCfg(node.Body, semanticModel, default).Should().NotBeNull();
+            ExtensionsCS.CreateCfg(node, semanticModel, default).Should().NotBeNull();
+        }
+
+        [TestMethod]
+        public void CreateCfg_PropertyDeclartion_ReturnsCfg_CS()
+        {
+            const string code = """
+                public class Sample
+                {
+                    int field;
+                    public int Property => field + 42;
+                }
+                """;
+            var (tree, semanticModel) = TestHelper.CompileCS(code);
+            var node = tree.Single<SyntaxCS.PropertyDeclarationSyntax>();
+
+            ExtensionsCS.CreateCfg(node, semanticModel, default).Should().NotBeNull();
+        }
+
+        [TestMethod]
+        public void CreateCfg_PropertyDeclartionWithoutExpressionBody_ReturnsNull_CS()
+        {
+            const string code = """
+                public class Sample
+                {
+                    public int Property {get; set;}
+                }
+                """;
+            var (tree, semanticModel) = TestHelper.CompileCS(code);
+            var node = tree.Single<SyntaxCS.PropertyDeclarationSyntax>();
+
+            ExtensionsCS.CreateCfg(node, semanticModel, default).Should().BeNull();
```

Comment:
This fragment seems duplicated many times. It could be `=> ValidateCreateCfg("""...` instead

Reply:
Not so trivial - the syntax type based on which we get the node for the cfg creation changes + the extensions; ' namespace  (depending if you want a cfg for CS or VB code).

I'll be happy to do this extraction in another PR.

Reply:
This PR starts being big and long-lived - I"ll follow up with another PR. (added a task to myself)

Reply:
https://github.com/SonarSource/sonar-dotnet/pull/8355